controller/webhook/launcher.go

```diff
@@ -0,0 +1,64 @@
+package webhook
+
+import (
+	"flag"
+	"os"
+	"os/signal"
+	"strconv"
+
+	"github.com/linkerd/linkerd2/controller/k8s"
+	"github.com/linkerd/linkerd2/pkg/admin"
+	"github.com/linkerd/linkerd2/pkg/flags"
+	"github.com/linkerd/linkerd2/pkg/tls"
+	log "github.com/sirupsen/logrus"
+)
+
+// Launch sets up and starts the webhook and metrics servers
+func Launch(config *Config) {
+	p := strconv.FormatUint(uint64(config.MetricsPort), 10)
+	metricsAddr := flag.String("metrics-addr", ":"+p, "address to serve scrapable metrics on")
+	addr := flag.String("addr", ":8443", "address to serve on")
+	kubeconfig := flag.String("kubeconfig", "", "path to kubeconfig")
+	controllerNamespace := flag.String("controller-namespace", "linkerd", "namespace in which Linkerd is installed")
+	flags.ConfigureAndParse()
+
+	stop := make(chan os.Signal, 1)
+	defer close(stop)
+	signal.Notify(stop, os.Interrupt, os.Kill)
+
+	k8sAPI, err := k8s.InitializeAPI(*kubeconfig, k8s.NS, k8s.RS)
```

Comment:
are `k8s.NS, k8s.RS` needed for all webhooks or just `proxy-injector`? if the latter, consider passing these in as part of the config struct.

Reply:
Yes, good catch :eye: